### ISSUE TITLE
Polylang not compatible with XML Sitemap & Google News, error of redirection

### ISSUE THREAD

**Issue by jcvignoli** [polylang/polylang#667]
Hi,

I just began to use Polylang 2.9 runing on wordpress 5.6.

Unfortunately, it breaks XML Sitemap & Google News plugin: the xml sitemap becomes unavailable, redirecting to another page of the website.

In concrete terms,
https://www.jcvignoli.com/blog/sitemap-news.xml
is redirected to
https://www.jcvignoli.com/blog/category/activisme

I also contacted XML Sitemap & Google News plugin developer on the wordpress forum.

**Comment by Chrystll** [comment]
I don't reproduce your issue with the 2.9 version of Polylang and the 5.2.7 version of https://wordpress.org/plugins/xml-sitemap-feed/. But I maybe missed a step. Could you check my steps and let me know what I'm missing?
1. My URL modifications is set from the directories as yours:
![screenshot-trunk-test com-2020 12 16-15_59_34](https://user-images.githubusercontent.com/19824724/102365868-4f2fd800-3fb8-11eb-8f77-bb96b369d448.png)

2. Here are my permalinks:
![screenshot-trunk-test com-2020 12 16-16_00_09](https://user-images.githubusercontent.com/19824724/102365887-5525b900-3fb8-11eb-819a-be7d0b15fdc1.png)

3. My homepage displays the latest posts as yours:
![screenshot-trunk-test com-2020 12 16-16_00_09 (1)](https://user-images.githubusercontent.com/19824724/102365922-5ce55d80-3fb8-11eb-9a8d-62dc8c30ad54.png)

4. When clicking on the "View" link of the XML Sitemap Index (see below) I'm well redirecting to the http://trunk-test.com/sitemap.xml without any redirection
![screenshot-trunk-test com-2020 12 16-16_03_19 (1)](https://user-images.githubusercontent.com/19824724/102366024-75ee0e80-3fb8-11eb-8ebc-8e8d5798b3f7.png)

![screenshot-trunk-test com-2020 12 16-16_03_19 (2)](https://user-images.githubusercontent.com/19824724/102366088-84d4c100-3fb8-11eb-8e39-86ef11fbaeb0.png)

**Comment by jcvignoli** [comment]
Hi,

Could you try to activate "Google News Sitemap" instead of "XML Sitemap Index" and visit yourwebsite.com/sitemap-news.xml ?

**Comment by Chrystll** [comment]
Hello,

I don't reproduce your redirection when activating "Google News Sitemap" instead of "XML Sitemap Index". I get yourwebsite.com/sitemap-news.xml as expected (see the attached video). 
[Kazam_screencast_00020.webm.zip](https://github.com/polylang/polylang/files/5716182/Kazam_screencast_00020.webm.zip)

I kept the options by default of the XML Sitemap & Google News plugins, see:

![screenshot-trunk-test com-2020 12 18-15_21_05](https://user-images.githubusercontent.com/19824724/102624873-ea55b880-4144-11eb-9d58-fa70b990c0d1.png)

![screenshot-trunk-test com-2020 12 18-15_21_45](https://user-images.githubusercontent.com/19824724/102624882-efb30300-4144-11eb-9d40-eaef61502434.png)

**Comment by jcvignoli** [comment]
Thanks for your response! but frankly, I'm lost.

I discussed the matter with XML Sitemap & Google News' author, who could reproduce it and provided a developement version that takes care of the issue.

I then uninstalled the developement version (5.3.0-alpha-15) and installed the current version (5.2.7) on a test website, and the problem disappeared.

If I'm able to reproduce the issue, I'll let you know.
